### PR TITLE
Refine the query for the select all records operation

### LINKED ISSUE #1167: Add select all items in record list (includes records on all pages)
Add the capability to select all records in the record list from all pages, and allow bulk editing/deleting/exporting/automation and other things you can do with selected records.

### PR DESCRIPTION
**Pull request by KinyaElGrande**
Refine the query logic used for the select all records feature in the RecordList component, specifically for bulk edit, delete, and undelete operations.

### CODE REVIEW

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 1]
```diff
@@ -1555,25 +1555,23 @@ export default {
           }
         }
       } else {
-        // Pick module and namespace ID from the first record
-        //
-        // We are always showing list of records from the
-        // same module so this should be safe to do.
-        const { moduleID, namespaceID } = this.items[0].r
-
-        let query = ''
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue:1564]: let query = this.filter.query

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 2]
```diff
@@ -639,6 +640,7 @@
         :selected-records="inlineEdit.recordIDs"
         :selected-fields="inlineEdit.fields"
         :initial-record="inlineEdit.record"
+        :filter-query="filter.query"
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue:643]: rename from filter-query to query

File: client/web/compose/src/components/Public/Record/BulkEdit/index.vue [hunk 3]
```diff
@@ -179,6 +184,10 @@ export default {
     getFieldSelectorPlaceholder () {
       return this.$t(`recordList.bulkRecord.field.add${this.fields.length ? 'Another' : ''}`)
     },
+
+    selectedQuery () {
+      return this.filterQuery + this.selectedRecords.map(id => `recordID=${id}`).join(' OR ')
```
Comment (Fajfa) [inline client/web/compose/src/components/Public/Record/BulkEdit/index.vue:189]: return `this.selectedRecords.length ? this.selectedRecords.map(id => `recordID=${id}`).join(' OR ') : this.query`

File: locale/en/corteza-webapp-compose/notification.yaml [hunk 4]
```diff
@@ -119,7 +119,7 @@ record:
   deleteBulkRecordUpdateFailed: Could not update selected records
   deleteBulkSuccess: Successfully deleted selected records
   bulkRecordUpdateSuccess: Successfully updated selected records
-  undeleteBulkFailed: Could not restore selected records
+  restoreBulkFailed: Could not restore selected records
```
Comment (Fajfa) [inline locale/en/corteza-webapp-compose/notification.yaml:122]: Fix undeleteFailed aswell please

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 5]
```diff
@@ -196,6 +196,7 @@
               :module="recordListModule"
               :namespace="namespace"
               :selected-records="selectedAllRecords ? [] : selected"
+              :filter-query="filter.query"
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue]: :query="bulkQuery"

bulkQuery is a computed that is used for all bulks

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 6]
```diff
@@ -196,6 +196,7 @@
               :module="recordListModule"
               :namespace="namespace"
               :selected-records="selectedAllRecords ? [] : selected"
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue]: Remove the selected records prop and replace it with query

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 7]
```diff
@@ -1555,25 +1555,23 @@ export default {
...
+          this.filter.query = this.filter.query + recordIDs.map(r => `recordID='${r}'`).join(' OR ')
         }
 
         this.processing = true
 
+        // Pick module and namespace ID from the filter
+        //
+        // We are always showing list of records from the
+        // same module so this should be safe to do.
+        const { moduleID, namespaceID, query } = this.filter
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue]: Keep the moduleID and namepsaceID but get query from this.bulkQuery

File: client/web/compose/src/components/Public/Record/BulkEdit/index.vue [hunk 8]
```diff
@@ -88,7 +88,7 @@
             <b-button
               variant="primary"
               :disabled="!fields.length || processing"
-              @click="handleBulkUpdateSelectedRecords(selectedRecords)"
+              @click="handleBulkUpdateSelectedRecords(selectedQuery)"
```
Comment (Fajfa) [inline client/web/compose/src/components/Public/Record/BulkEdit/index.vue]: use query from prop

File: client/web/compose/src/components/Public/Record/BulkEdit/index.vue [hunk 9]
```diff
@@ -156,6 +156,11 @@ export default {
       type: String,
       default: '',
     },
+
+    filterQuery: {
```
Comment (Fajfa) [inline client/web/compose/src/components/Public/Record/BulkEdit/index.vue]: Rename this to query and remove the selected records prop

File: client/web/compose/src/components/Public/Record/BulkEdit/index.vue [hunk 10]
```diff
@@ -179,6 +184,10 @@ export default {
     getFieldSelectorPlaceholder () {
       return this.$t(`recordList.bulkRecord.field.add${this.fields.length ? 'Another' : ''}`)
     },
+
+    selectedQuery () {
```
Comment (Fajfa) [inline client/web/compose/src/components/Public/Record/BulkEdit/index.vue]: Remove this

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 11]
```diff
@@ -639,6 +640,7 @@
         :selected-records="inlineEdit.recordIDs"
         :selected-fields="inlineEdit.fields"
         :initial-record="inlineEdit.record"
+        :filter-query="filter.query"
         :modal-title="$t('recordList.inlineEdit.modal.title')"
         open-on-select
         @save="onInlineEdit()"
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue:644]: Add `@close="onInlineEditClose()"`

File: client/web/compose/src/components/PageBlocks/RecordListBase.vue [hunk 12]
```diff
@@ -1555,25 +1555,23 @@ export default {
...
         this.processing = true
 
+        // Pick module and namespace ID from the filter
+        //
+        // We are always showing list of records from the
+        // same module so this should be safe to do.
+        const { moduleID, namespaceID, query } = this.filter
+
         this.$ComposeAPI
           .recordBulkDelete({ moduleID, namespaceID, query })
```
Comment (Fajfa) [inline client/web/compose/src/components/PageBlocks/RecordListBase.vue:1566]:
```
    editInlineField (record, field) {
      this.inlineEdit.fields = [field]
      this.inlineEdit.record = record.clone()
      this.inlineEdit.query = `recordID = ${record.recordID}`
    },

    onInlineEditClose () {
      this.inlineEdit.fields = []
      this.inlineEdit.record = {}
      this.inlineEdit.query = ''
    },

    onInlineEdit () {
      this.refresh(true)
      this.onInlineEditClose()
    },
```

File: client/web/compose/src/components/Public/Record/BulkEdit/index.vue [hunk 13]
```diff
@@ -179,6 +184,10 @@ export default {
...
+      return this.filterQuery + this.selectedRecords.map(id => `recordID=${id}`).join(' OR ')
+    },
   },
 
   watch: {
```
Comment (Fajfa) [inline client/web/compose/src/components/Public/Record/BulkEdit/index.vue:184]: Replace selectedRecords watcher with query

```
    query: {
      handler (query) {
        if (!this.openOnSelect || !query.length) return

        this.record = new compose.Record(this.module, this.initialRecord)
        this.showModal = true
      },
    },
```

and add `this.$emit('close')` to the end of onModalHide

```
    onModalHide () {
      this.showModal = false

      if (this.openOnSelect) {
        this.fields = []
        this.record = new compose.Record(this.module, {})
      }
      this.$emit('close')
    },
```

**Review by Fajfa — CHANGES_REQUESTED** [review]

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
LGTM

**Pull request merged**